tests/unit/test_platforms_get_platform.py

```diff
@@ -237,18 +237,28 @@ def test_get_platform_using_platform_name_from_job_info(
 
 
 def test_get_platform_groups_basic(mock_glbl_cfg):
-    # get platform from group works.
+    """get platform from group works.
+
+    Additionally, ensure that we stop after selecting the first
+    appropriate platform.
+    """
     mock_glbl_cfg(
         'cylc.flow.platforms.glbl_cfg',
         '''
         [platforms]
-            [[aleph]]
-                hosts = aleph
-            [[bet]]
-                hosts = bet
+            [[aleph, bet, alpha, beta]]
 
         [platform groups]
             [[hebrew_letters]]
+                platforms = alpha, beta
+                [[[selection]]]
+                    method = definition order
+            [[aleph]]
+            # Group with same name as platform to try and
+            # trip up the platform selection logic after it
+            # has processed [[.*_letters]] below
```

Comment:
Yes. That's a good idea, this test is wierd and non obvious
Feel free to add the following if you like.
```suggestion
            [[aleph]]
            # Group with same name as platform to try and
            # trip up the platform selection logic after it
            # has processed [[.*_letters]] below
            # https://github.com/cylc/cylc-flow/pull/5398#discussion_r1130692633
```